src/Avalonia.Animation/Keyframes/KeyFrames`1.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Avalonia.Collections;
+using System.ComponentModel;
+using Avalonia.Animation.Utils;
+using System.Reactive.Linq;
+using System.Linq;
+using Avalonia.Data;
+using System.Reactive.Disposables;
+
+namespace Avalonia.Animation.Keyframes
+{
+    /// <summary>
+    /// Base class for KeyFrames objects
+    /// </summary>
+    public abstract class KeyFrames<T> : AvaloniaList<KeyFrame>, IKeyFrames
+    {
+
+        /// <summary>
+        /// Target property.
+        /// </summary>
+        public AvaloniaProperty Property { get; set; }
+
+        /// <summary>
+        /// List of type-converted keyframes.
+        /// </summary>
+        public Dictionary<double, T> ConvertedKeyframes = new Dictionary<double, T>();
+
+        private bool IsVerfifiedAndConverted;
+
+
+        /// <inheritdoc/>
+        public virtual IDisposable Apply(Animation animation, Animatable control, IObservable<bool> obsMatch)
+        {
+            if (!IsVerfifiedAndConverted)
```

Comment:
Do we want to invalidate this flag if keyframes are added/removed after `VerifyConvertKeyFrames` is called? Or should we just not support that?